src/semidiscretization/semidiscretization_hyperbolic_parabolic.jl

```diff
@@ -0,0 +1,235 @@
+# By default, Julia/LLVM does not use fused multiply-add operations (FMAs).
+# Since these FMAs can increase the performance of many numerical algorithms,
+# we need to opt-in explicitly.
+# See https://ranocha.de/blog/Optimizing_EC_Trixi for further details.
+@muladd begin
+
+
+"""
+    SemidiscretizationHyperbolicParabolic
+
+A struct containing everything needed to describe a spatial semidiscretization
+of a mixed hyperbolic-parabolic conservation law.
+"""
+struct SemidiscretizationHyperbolicParabolic{Mesh, Equations, ParabolicEquations, InitialCondition,
+                                             BoundaryConditions, ParabolicBoundaryConditions,
+                                             SourceTerms, Solver, Cache, ParabolicCache} <: AbstractSemidiscretization
+
+  mesh::Mesh
+
+  equations::Equations
+  parabolic_equations::ParabolicEquations
+
+  # This guy is a bit messy since we abuse it as some kind of "exact solution"
+  # although this doesn't really exist...
+  initial_condition::InitialCondition
+
+  boundary_conditions::BoundaryConditions
+  parabolic_boundary_conditions::ParabolicBoundaryConditions
+
+  source_terms::SourceTerms
+
+  solver::Solver
+  # TODO: do we want to introduce `parabolic_solver` for future specialization?
+
+  cache::Cache
+  parabolic_cache::ParabolicCache
+
+  performance_counter::PerformanceCounter
+
+  function SemidiscretizationHyperbolicParabolic{Mesh, Equations, ParabolicEquations, InitialCondition, BoundaryConditions, ParabolicBoundaryConditions, SourceTerms, Solver, Cache, ParabolicCache}(
+      mesh::Mesh, equations::Equations, parabolic_equations::ParabolicEquations, initial_condition::InitialCondition,
+      boundary_conditions::BoundaryConditions, parabolic_boundary_conditions::ParabolicBoundaryConditions,
+      source_terms::SourceTerms, solver::Solver, cache::Cache, parabolic_cache::ParabolicCache) where {Mesh, Equations, ParabolicEquations, InitialCondition, BoundaryConditions, ParabolicBoundaryConditions, SourceTerms, Solver, Cache, ParabolicCache}
+    @assert ndims(mesh) == ndims(equations)
+
+    # Todo: assert nvariables(equations)==nvariables(parabolic_equations)
+
+    performance_counter = PerformanceCounter()
+
+    new(mesh, equations, parabolic_equations, initial_condition,
+        boundary_conditions, parabolic_boundary_conditions,
+        source_terms, solver, cache, parabolic_cache, performance_counter)
+  end
+end
+
+"""
+    SemidiscretizationHyperbolicParabolic(mesh, equations, initial_condition, solver;
+                                  source_terms=nothing,
+                                  boundary_conditions=boundary_condition_periodic,
+                                  RealT=real(solver),
+                                  uEltype=RealT,
+                                  initial_cache=NamedTuple())
+
+Construct a semidiscretization of a hyperbolic PDE.
+"""
+function SemidiscretizationHyperbolicParabolic(mesh, equations, parabolic_equations,
```

Comment:
I think that's a good idea - to emphasize the equations and BCs go together. I'll update the PR with that soon.

I think the merged approach will be difficult because of the dispatch on parabolic equation type, but I also haven't thought this through.

Reply:
What about a Navier-Stokes system. I guess we still need to know the underlying Euler system when computing some stuff, say, the temperature gradients? How shall we pass that to the parabolic part? Can we make sure that the ratio of specific heats of the hyperbolic and parabolic parts fit together or is this up to the user to check? Can we re-use functions like `temperature(u, equations::CompressibleEulerEquations)` for this?

Reply:
I guess we should just try to see how it works with the separated approach and then see if we can hack it together with a merged system? I see benefits in both cases: Having a single system is easier conceptually for a user, but if we have, eg., an Euler system and a Navier-Stokes system, there's obviously going to be a lot of overhead as well.

Reply:
Yes; you do need to know the underlying Euler part for Navier-Stokes, but you could (for example) have the Euler equations as a field in the Navier-Stokes type. 

A merged type could also dispatch to the hyperbolic and parabolic type individually, and just be used to ensure consistency when constructing the two equations.

Reply:
Agreed. In any case, as long as this is marked experimental, we are free to change it later. I consider this decision to be (mostly) an implementation detail, since I think both (or all three?) approaches will work, and thus we can postpone the final decision to the "make it nice" phase.